modules/passkey/test/SafeWebAuthnSignerFactory.spec.ts

```diff
@@ -0,0 +1,178 @@
+import { expect } from 'chai'
+import { deployments, ethers } from 'hardhat'
+import { DUMMY_AUTHENTICATOR_DATA, base64UrlEncode, encodeWebAuthnSigningMessage, getSignatureBytes } from './utils/webauthn'
+
+describe('SafeWebAuthnSignerFactory', () => {
+  const setupTests = deployments.createFixture(async () => {
+    const { SafeWebAuthnSignerFactory } = await deployments.fixture()
+    const factory = await ethers.getContractAt('SafeWebAuthnSignerFactory', SafeWebAuthnSignerFactory.address)
+
+    const MockContract = await ethers.getContractFactory('MockContract')
+    const mockVerifier = await MockContract.deploy()
+
+    return { factory, mockVerifier }
+  })
+
+  describe('getSigner', function () {
+    it('Should return the address that a signer will be created on', async () => {
+      const { factory, mockVerifier } = await setupTests()
+
+      const x = ethers.id('publicKey.x')
+      const y = ethers.id('publicKey.y')
+
+      const signer = await factory.getSigner(x, y, mockVerifier)
+
+      expect(ethers.dataLength(await ethers.provider.getCode(signer))).to.equal(0)
+
+      await factory.createSigner(x, y, mockVerifier)
+
+      expect(ethers.dataLength(await ethers.provider.getCode(signer))).to.not.equal(0)
+    })
+
+    it('Should return different signer for different inputs', async () => {
+      const { factory, mockVerifier } = await setupTests()
+
+      const x = ethers.id('publicKey.x')
+      const y = ethers.id('publicKey.y')
+
+      const signer = await factory.getSigner(x, y, mockVerifier)
+
+      for (const params of [
+        [ethers.id('publicKey.otherX'), y, mockVerifier],
+        [x, ethers.id('publicKey.otherY'), mockVerifier],
+        [x, y, '0x0000000000000000000000000000000000000100'],
+      ] as const) {
+        expect(await factory.getSigner(...params)).to.not.equal(signer)
+      }
+    })
+  })
+
+  describe('createSigner', function () {
+    it('Should create a signer and return its address', async () => {
```

Comment:
Similar to the previous comment, we ensure the creation is successful, doesn't check the `return` value.

Reply:
Changed test to check deterministic deployment address.